server/work-queue/jobs/extract-page.js

```diff
@@ -0,0 +1,36 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+var log = require('../../logger')('server.work-queue.jobs.extract-url');
+var userPage = require('../../models/user-page');
+var embedly = require('../../embedly');
+
+module.exports = {
+  // o is an object with keys { userId, url, urlHash, title }
+  perform: function(o, cb) {
+    log.verbose('job created with params ' + JSON.stringify(o));
+    embedly.extract(o.url, function(err, data) {
```

Comment:
maybe we can group embedly.js, postgres.js, elasticsearch.js, and queue.js under "things which require network IO", and create some kind of unified interface/contract for all of them. Not just promises or callbacks, but an API and some standard object properties. (Wouldn't it be nice for __heartbeat__ to just do something like: "ask each of the Service objects if they are able to connect"?